bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/Item.java

```diff
@@ -77,7 +77,7 @@
      *
      * @return a list of data types that can be used to update the item state
      */
-    public List<Class<? extends State>> getAcceptedDataTypes();
+    public @NonNull List<@NonNull Class<@NonNull ? extends State>> getAcceptedDataTypes();
```

Comment:
Maybe stupid question: What does `Class<@NonNull ? extends State>`tell me?

Reply:
Okay, perhaps for Class this is really not necessary.
It would state that if the generic template type T (`? extends State`) is used e.g. on some methods return value, you already state that in this case the type is non null.
So, `T cast(Object obj)` would return a non-null type / same for newInstance.
But it is not only not necessary, but also wrong in this case.

> *haha*

Is a failing test such funny? :wink:

Reply:
>> haha

> Is a failing test such funny? 😉

No. Just a (bad) attempt of a Nelson (from The Simpsons) cite.

Reply:
PS ...and thanks for the explanation!